toolchain/driver/compile_subcommand.cpp

```diff
@@ -890,16 +883,18 @@ auto CompileSubcommand::Run(DriverEnv& driver_env) -> DriverResult {
     }
   }
 
-  // Prepare CompilationUnits before building scope exit handlers.
-  llvm::SmallVector<std::unique_ptr<CompilationUnit>> units;
-  int unit_index = -1;
-  auto unit_builder = [&](llvm::StringRef filename) {
-    return std::make_unique<CompilationUnit>(
-        ++unit_index, &driver_env, &options_, &driver_env.consumer, filename);
-  };
-  llvm::append_range(units, llvm::map_range(prelude, unit_builder));
-  llvm::append_range(units,
-                     llvm::map_range(options_.input_filenames, unit_builder));
+  // Prepare CompilationUnits before building scope exit handlers. Prelude files
+  // are first, then input files.
+  llvm::SmallVector<std::unique_ptr<CompilationUnit>> units(llvm::map_range(
+      llvm::seq(prelude.size() + options_.input_filenames.size()),
+      [&](size_t i) -> std::unique_ptr<CompilationUnit> {
+        llvm::StringRef filename =
+            (i < prelude.size()) ? llvm::StringRef(prelude[i])
+                                 : options_.input_filenames[i - prelude.size()];
+        return std::make_unique<CompilationUnit>(
+            SemIR::CheckIRId(i), &driver_env, &options_, &driver_env.consumer,
+            filename);
+      }));
```

Comment:
I find this way of structuring the code harder to follow. Would it work to stick with the previous code, and just change `++unit_index` to `SemIR::CheckIRId(++unit_index)`?